beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix_test.go

```diff
@@ -850,9 +888,19 @@ func TestServer_getPayloadHeader(t *testing.T) {
 	}
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			vs := &Server{BlockBuilder: tc.mock, HeadFetcher: tc.fetcher, TimeFetcher: &blockchainTest.ChainService{
+			vs := &Server{BeaconDB: dbTest.SetupDB(t), BlockBuilder: tc.mock, HeadFetcher: tc.fetcher, TimeFetcher: &blockchainTest.ChainService{
 				Genesis: genesis,
 			}}
+			require.NoError(
+				t, vs.BeaconDB.SaveRegistrationsByValidatorIDs(context.Background(), []primitives.ValidatorIndex{0}, []*ethpb.ValidatorRegistrationV1{
```

Comment:
hmm what about using the cache instead? 
i think you can do tc.mock.RegistrationCache = NewRegistrationCache() // you'd probably need to initialize before hand and add those values.